src/test/java/stashpullrequestbuilder/stashpullrequestbuilder/StashRepositoryTest.java

```diff
@@ -214,4 +244,70 @@ public void getTargetPullRequestsBuildsIfSkipPhraseIsNull() {
 
     assertThat(stashRepository.getTargetPullRequests(), contains(pullRequest));
   }
+
+  @Test
+  public void getAdditionalParametersUsesNewestParameterDefinitionForOldCommentsFirst() {
+    StashPullRequestComment comment1 = new StashPullRequestComment();
+    comment1.setCommentId(1);
+    comment1.setText("p:key=value1");
+
+    StashPullRequestComment comment2 = new StashPullRequestComment();
+    comment2.setCommentId(2);
+    comment2.setText("p:key=value2");
+
+    List<StashPullRequestComment> comments = Arrays.asList(comment1, comment2);
+    when(stashApiClient.getPullRequestComments(any(), any(), any())).thenReturn(comments);
+
+    Map<String, String> parameters = stashRepository.getAdditionalParameters(pullRequest);
+
+    assertThat(parameters, is(aMapWithSize(1)));
+    assertThat(parameters, hasEntry("key", "value2"));
+  }
+
+  @Test
+  public void getAdditionalParametersUsesNewestParameterDefinitionForNewCommentsFirst() {
```

Comment:
I think the names might be a little confusing here. I would name the previous case `getAdditionalParametersUsesNewestParameterDefinition` and this one `getAdditionalParametersUsesNewestParameterDefinitionRegardlessOfListOrder`

Reply:
Done.